napari/components/viewer_model.py

```diff
@@ -458,8 +458,9 @@ def _on_grid_change(self, event):
         if self._block_repaint:
             return
         extent = self._sliced_extent_world
-        for i, layer in enumerate(self.layers[::-1]):
-            i_row, i_column = self.grid.position(i, len(self.layers))
+        n_layers = len(self.layers)
+        for i, layer in enumerate(self.layers):
+            i_row, i_column = self.grid.position(n_layers - 1 - i, n_layers)
```

Comment:
why no simple `reversed` use? It is generator

Reply:
i did it because `self.layers` is one of our custom evented lists and so emits events during operations like this. i thought there might be a speed up here, but it actually doesn't look like it. it's a change i wanted to make anyway as part of some improvements to the layerlist i have in another PR, but i can save it till then if you prefer